src/cursor/cur_index.c

```diff
@@ -210,11 +240,16 @@ __curindex_search(WT_CURSOR *cursor)
 	 * otherwise the primary key columns will be appended to the index key,
 	 * but we don't disallow that (odd) case.
 	 */
-	if (exact < 0)
+	if (cmp < 0)
 		WT_ERR(child->next(child));
 
-	if (child->key.size < cursor->key.size ||
-	    memcmp(child->key.data, cursor->key.data, cursor->key.size) != 0) {
+	if (child->key.size < cursor->key.size)
+		WT_ERR(WT_NOTFOUND);
```

Comment:
This check was already there but I had to page it in as well.  We're going to compare a prefix of the key (just the index key) to make sure it matches.  If the key we found is shorter than the prefix we're comparing with, we're done: there's no way it can be equal.

I'll add a comment to clarify.